cpp/src/arrow/engine/substrait/function_test.cc

```diff
@@ -188,6 +188,12 @@ TEST(FunctionMapping, ValidCases) {
        {int8(), int8()},
        "-119",
        int8()},
+      {{kArrowSimpleExtensionFunctionsUri, "add_checked"},
```

Comment:
I wonder if this also works for UDFs? e.g., if I register a Python UDF by name "my_func", I can also use the same `::arrow::compute::call` to invoke it via `kSimpleSubstraitToArrow` or is there more to it?

Reply:
It should work for any scalar UDF used in a project node or filter node.  It would also work for aggregate UDFs (though these cannot be specified in python).

`::arrow::compute::call` is evaluated by looking up the function, by name, in the function registry given to the plan.  So the UDF will either need to be added to the default function registry or the custom function registry will need to be specified when the plan is run (I believe pyarrow.substrait.run_query exposes this configuration property but, if not, we can add it easily).

Reply:
thx for the explaination